agents/official/beeai-framework/chat/src/chat/agent.py

```diff
@@ -124,49 +149,100 @@ async def chat(message: Message, context: Context):
     The agent is an AI-powered conversational system with memory, supporting real-time search, Wikipedia lookups,
     and weather updates through integrated tools.
     """
-    # ensure the model is pulled before running
-    os.environ["OPENAI_API_BASE"] = os.getenv("LLM_API_BASE", "http://localhost:11434/v1")
-    os.environ["OPENAI_API_KEY"] = os.getenv("LLM_API_KEY", "dummy")
-    llm = ChatModel.from_name(
-        f"openai:{os.getenv('LLM_MODEL', 'llama3.1')}",
-        ChatModelParameters(temperature=0),
+    extracted_files = await extract_files(
+        history=messages[context.context_id], incoming_message=message
     )
+    input = to_framework_message(message)
 
     # Configure tools
-    tools: list[AnyTool] = [
+    tools = [
+        ActTool(),
```

Comment:
can you clarify what is this tool for? It's a tool called before other tools are called?

Reply:
I've added some comments